changelog.d/7961.bugfix

```diff
@@ -0,0 +1 @@
+Fix sometimes returning 500 instead of the actual error response when using opentracing.
```

Comment:
```suggestion
Fix a bug which could cause `500 Internal Server Error` to be returned instead of the actual response when an error was returned from REST APIs and opentracing was enabled..
```